### PR TITLE
Hooks/button

### PR DESCRIPTION
**Pull request by talesdsp**
This PR converts Button from Class to Hooks.
All tests passing.

### CODE REVIEW

File: src/__tests__/Button.test.js [hunk 1]
```diff
@@ -267,39 +310,52 @@ describe("<Button />", () => {
...
+   */
+
   it("ButtonType: highlight", () => {
-    const component = shallow(<Button highlight />);
-    expect(component.instance().props.highlight).toEqual(true);
-    expect(component.name()).toEqual("TouchableHighlight");
-  });
+    const component = shallow(
+      <Button highlight>
+        <Button />
```
Comment (hetmann) [inline src/__tests__/Button.test.js]: We can add a Text here :) Button in Button is kinda weird and can lead to events conflict. Think about which button should you set the onPress.

File: src/__tests__/Button.test.js [hunk 2]
```diff
@@ -88,16 +92,22 @@ describe("<Button />", () => {
   });
 
   it('color="#88B04B"', () => {
-    const component = renderer.create(<Button color="#88B04B" />).toJSON();
-    const instance = renderer.create(<Button color="#88B04B" />).getInstance();
+    const B = renderer.create(<Button color="#88B04B" />);
```
Comment (hetmann) [inline src/__tests__/Button.test.js]: I like then variable names are self explanatory. B could be Button but it conflicts with the component name, then we can name it something more relevant, maybe: button or something else :)

**Review by hetmann — COMMENTED** [review]
@taleco22 Thank you for the PR. I just left a few comments

**Comment by talesdsp** [comment]
Fine, will fix those in a moment.

**Pull request merged**